doc/source/rllib/core-concepts.rst

```diff
@@ -42,39 +98,237 @@ define a trainable policy with a functional-style API, for example:
+        for w in workers.remote_workers():
+            w.set_weights.remote(weights)
+
 Sample Batches
 --------------
 
-Whether running in a single process or `large cluster <rllib-training.html#specifying-resources>`__, all data interchange in RLlib is in the form of `sample batches <https://github.com/ray-project/ray/blob/master/rllib/policy/sample_batch.py>`__. Sample batches encode one or more fragments of a trajectory. Typically, RLlib collects batches of size ``rollout_fragment_length`` from rollout workers, and concatenates one or more of these batches into a batch of size ``train_batch_size`` that is the input to SGD.
+Whether running in a single process or a `large cluster <rllib-training.html#specifying-resources>`__,
+all data in RLlib is interchanged in the form of `sample batches <https://github.com/ray-project/ray/blob/master/rllib/policy/sample_batch.py>`__.
+Sample batches encode one or more fragments of a trajectory.
+Typically, RLlib collects batches of size ``rollout_fragment_length`` from rollout workers, and concatenates one or
+more of these batches into a batch of size ``train_batch_size`` that is the input to SGD.
 
-A typical sample batch looks something like the following when summarized. Since all values are kept in arrays, this allows for efficient encoding and transmission across the network:
+A typical sample batch looks something like the following when summarized.
+Since all values are kept in arrays, this allows for efficient encoding and transmission across the network:
 
 .. code-block:: python
 
- { 'action_logp': np.ndarray((200,), dtype=float32, min=-0.701, max=-0.685, mean=-0.694),
-   'actions': np.ndarray((200,), dtype=int64, min=0.0, max=1.0, mean=0.495),
-   'dones': np.ndarray((200,), dtype=bool, min=0.0, max=1.0, mean=0.055),
-   'infos': np.ndarray((200,), dtype=object, head={}),
-   'new_obs': np.ndarray((200, 4), dtype=float32, min=-2.46, max=2.259, mean=0.018),
-   'obs': np.ndarray((200, 4), dtype=float32, min=-2.46, max=2.259, mean=0.016),
-   'rewards': np.ndarray((200,), dtype=float32, min=1.0, max=1.0, mean=1.0),
-   't': np.ndarray((200,), dtype=int64, min=0.0, max=34.0, mean=9.14)}
+    sample_batch = { 'action_logp': np.ndarray((200,), dtype=float32, min=-0.701, max=-0.685, mean=-0.694),
+        'actions': np.ndarray((200,), dtype=int64, min=0.0, max=1.0, mean=0.495),
+        'dones': np.ndarray((200,), dtype=bool, min=0.0, max=1.0, mean=0.055),
+        'infos': np.ndarray((200,), dtype=object, head={}),
+        'new_obs': np.ndarray((200, 4), dtype=float32, min=-2.46, max=2.259, mean=0.018),
+        'obs': np.ndarray((200, 4), dtype=float32, min=-2.46, max=2.259, mean=0.016),
+        'rewards': np.ndarray((200,), dtype=float32, min=1.0, max=1.0, mean=1.0),
+        't': np.ndarray((200,), dtype=int64, min=0.0, max=34.0, mean=9.14)
+    }
 
 In `multi-agent mode <rllib-concepts.html#policies-in-multi-agent>`__, sample batches are collected separately for each individual policy.
```

Comment:
.. and wrapped up together in a `MultiAgentBatch` serving as a container for the individual agents' SampleBatches.

Reply:
done